library/std/src/io/impls.rs

```diff
@@ -361,14 +361,15 @@ impl Write for &mut [u8] {
 impl<A: Allocator> Write for Vec<u8, A> {
     #[inline]
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        self.try_reserve(buf.len()).map_err(oom_io_error)?;
         self.extend_from_slice(buf);
         Ok(buf.len())
     }
 
     #[inline]
     fn write_vectored(&mut self, bufs: &[IoSlice<'_>]) -> io::Result<usize> {
         let len = bufs.iter().map(|b| b.len()).sum();
```

Comment:
It's a pre-existing problem, but isn't it weird how this function ignores usize wraparound in the sum?

Reply:
I guess now that it handles OOM, there's no excuse that the program will just die on overflow. I've added overflow-checked sum.